### ISSUE TITLE
Next Step Microgrant: 1PLANET EcoNFT Factory and Launchpad dApp

### ISSUE THREAD

**Issue by ClimateFutures01** [filecoin-project/devgrants#720]
### 1. What is your project? (max 100 words)
**Imagine NFTs that automatically lower carbon emissions every time they are transacted.** Such ecoNFTs were developed to convert a regular NFT into a carbon reduction and ecosystem restoration “bot”. The ecoNFT contains code which utilizes the 1PLANET carbon credit token (1PLCO2) to automatically offset more carbon than the NFT creates from the blockchain network. This creates a carbon drawdown effect by supporting regenerative climate projects every time an ecoNFT is minted and resold. In other words, these are “carbon negative” NFTs that just keep reducing carbon seamlessly as collectors mint and trade them. Our MVP ecoNFT Launchpad allows anyone to create and deploy ecoNFT collections on the Polygon blockchain.

https://econft-ui.vercel.app   
(later it will use http://econfts.io)

### 2. How will IPFS, Filecoin, or related technology be used for this project? (max 100 words)
Our vision is to transform blockchain networks into decentralized “carbon drawdown networks” using NFTs as the vehicle and Filecoin  is instrumental and a core part of the ecoNFT collection creation process. The ecoNFT Launchpad dApp uses NFT.storage to save the ecoNFT content and json metadata to IPFS and Filecoin.

  
### 3. How will you improve your project with this grant? What steps will you take to meet this objective? (max 200 words)
We want to further develop the MVP by adding more metadata options to the current ecoNFT creation process. Different NFT marketplaces have different options for metadata and our current version was limited to the basics. We'd like to add more metadata categories/options to create more complex jsons, e.g. add in options for geo-coordinates, QR codes, and hopefully dynamic NFTs. This requires more complex coding in the dapp, new forms, and providing more types of ecoNFT/ERC721 implementation contracts for the users.
 
### 4. Is this project open source?
All of the smart contracts are open source and under MIT License. We plan to put the ecoNFT solidity code on NPM so that projects can more easily use the interface in custom NFT contracts (or other use cases). The launchpad dapp code isn't open source but we are open to discuss the best option/strategy on this.

### 5. Do you agree to share grant reports upon request, including a final grant report at the end of the three month period?
Absolutely!
  
### 6. Does your proposal comply with our Community Code of Conduct?
Yes

### 7. Links and submissions
https://econft-ui.vercel.app
https://www.climatefutures.io
  
### Additional questions:
* For each team member(s), please list name, email, Github account, and role in the project.
Jesse Uzzell, jesse@climatefutures.io, ClimateFutures01, project lead
Daniel Socek, daniel@climatefutures.io, dsocek, lead developer

* How did you learn about our microgrant program? 
Through people we met at Filecoin Foundation
* If your project was created as part of an event or hackathon:
The original concept for the ecoNFT was born at a Green NFT Hackathon last May 2021. However it was much further developed and refined as part of our recent self-funded development of the of the ecoNFT Factory and Launchpad MVP.
*Please link to your hackathon submission
https://github.com/ClimateFutures01/GreenMyNFT

**Comment by ErinOCon** [comment]
Hi @ClimateFutures01, thank you for your proposal! Can you confirm that the entire microgrant will be applied to work that is open source? This is a key point in our microgrant criteria. Also, can you provide a link to your project code?

**Comment by ClimateFutures01** [comment]
> Hi @ClimateFutures01, thank you for your proposal! Can you confirm that the entire microgrant will be applied to work that is open source? This is a key point in our microgrant criteria. Also, can you provide a link to your project code?

Your welcome @ErinOCon and thanks for contacting us. Yes, we confirm that the microgrant will be applied to content that will be made open source.

We had a discussion before with your colleagues at Filecoin Foundation regarding how that should work since our techstack many pieces.

The main solidity code for the ecoNFT factory and ecoNFTs are / will be open source but we just haven't prepared the github repo for the public yet -- its still an MVP / Beta and we wanted to add this additional microgrant funded functionality before we released it.

You're invited to it. The branches haven't been merged etc so as mentioned it's not ready yet for the public. We thought it was to be made public after we did the dev work from the grant.

https://github.com/ClimateFutures01/econft-factory

Let us know if you have more questions.
